dockers/Screenshotter/screenshotter.js

```diff
@@ -128,17 +130,53 @@ if (seleniumURL) {
     console.log("Selenium driver in local session");
 }
 
-if (!katexURL) {
-    katexURL = "http://" + katexIP + ":" + opts.katexPort + "/";
-}
 var toStrip = "http://localhost:7936/"; // remove this from testcase URLs
 
+process.nextTick(startServer);
+var attempts = 0;
+
+//////////////////////////////////////////////////////////////////////
+// Start up development server
+
+var devServer = null;
+var minPort = 32768;
+var maxPort = 61000;
+
+function startServer() {
+    if (katexURL || katexPort) {
+        process.nextTick(tryConnect);
+        return;
+    }
+    var onerror = function() {
+        if (++attempts > 50) {
+            throw new Error("Failed to start up dev server");
+        }
+        process.nextTick(startServer);
+    };
+    var port = Math.floor(Math.random() * (maxPort - minPort)) + minPort;
+    var server = http.createServer(app).listen(port);
+    server.once("error", onerror);
+    server.once("listening", function() {
+        server.removeListener("error", onerror);
+        devServer = server;
+        katexPort = port;
+        attempts = 0;
+        process.nextTick(tryConnect);
+    });
+}
+
 //////////////////////////////////////////////////////////////////////
 // Wait for container to become ready
 
-var attempts = 0;
-process.nextTick(seleniumIP ? tryConnect : buildDriver);
 function tryConnect() {
+    if (!katexURL) {
+        katexURL = "http://" + katexIP + ":" + katexPort + "/";
+        console.log("KaTeX URL is " + katexURL);
+    }
+    if (!seleniumIP) {
+        process.nextTick(buildDriver);
```

Comment:
Not really, might as well make that all `nextTick`. I guess that when initially writing the code, I might have been thinking about keeping things simpler by just using one function, but since I'm using `nextTick` in a lot of cases now, it's certainly the more appropriate choice.